src/globus_sdk/paging/base.py

```diff
@@ -1,9 +1,36 @@
 import abc
-from typing import Any, Callable, Dict, Iterable, Iterator, List, Optional, TypeVar
+import functools
+import inspect
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Generic,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Type,
+    TypeVar,
+    cast,
+)
+
+import typing_extensions as te
```

Comment:
This is the only thing which needs attention still, as far as I know. The PR is basically ready, but we should only import `ParamSpec` from typing-extensions on sufficiently old python versions.